plasma_framework/contracts/src/exits/payment/controllers/PaymentStartInFlightExit.sol

```diff
@@ -309,14 +270,39 @@ library PaymentStartInFlightExit {
 
     function setInFlightExitInputs(
         PaymentExitDataModel.InFlightExit storage ife,
-        PaymentTransactionModel.Transaction[] memory inputTxs,
-        UtxoPosLib.UtxoPos[] memory inputUtxosPos
+        StartExitData memory exitData
     )
         private
     {
-        for (uint i = 0; i < inputTxs.length; i++) {
-            uint16 outputIndex = inputUtxosPos[i].outputIndex();
-            ife.inputs[i] = inputTxs[i].outputs[outputIndex];
+        for (uint i = 0; i < exitData.inputTxs.length; i++) {
+            uint16 outputIndex = exitData.inputUtxosPos[i].outputIndex();
+            WireTransaction.Output memory output = WireTransaction.getOutput(exitData.inputTxs[i], outputIndex);
+
+            ife.inputs[i] = PaymentExitDataModel.WithdrawData(
+                output.outputGuard,
+                exitData.outputIds[i],
+                output.token,
+                output.amount
+            );
+        }
+    }
+
+    function setInFlightExitOutputs(
```

Comment:
haha okay, if we are setting outputs here, I will think again whether I want to separate the piggyback into two functions or not in PR #222 . I personal have preference over simplicity than optimization so I do like to set outputs here to make logic consistent. Let's see how the gas goes.